browser/brave_ads/brave_ads_host.cc

```diff
@@ -0,0 +1,175 @@
+/* Copyright (c) 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/browser/brave_ads/brave_ads_host.h"
+
+#include <memory>
+#include <string>
+#include <utility>
+
+#include "brave/browser/brave_ads/ads_service_factory.h"
+#include "brave/browser/brave_rewards/rewards_service_factory.h"
+#include "brave/browser/extensions/brave_component_loader.h"
+#include "brave/components/brave_ads/browser/ads_service.h"
+#include "brave/components/brave_rewards/browser/rewards_service.h"
+#include "chrome/browser/extensions/extension_service.h"
+#include "chrome/browser/profiles/profile.h"
+#include "chrome/browser/ui/browser.h"
+#include "chrome/browser/ui/browser_finder.h"
+#include "content/public/browser/web_contents.h"
+#include "extensions/browser/extension_system.h"
+#include "extensions/common/constants.h"
+
+namespace brave_ads {
+
+namespace {
+
+constexpr char kAdsEnableRelativeUrl[] = "request_ads_enabled_panel.html";
+
+}  // namespace
+
+BraveAdsHost::BraveAdsHost(Profile* profile, content::WebContents* web_contents)
+    : WebContentsObserver(web_contents),
+      profile_(profile),
+      web_contents_(web_contents) {
+  DCHECK(profile_);
+  DCHECK(web_contents_);
```

Comment:
I guess you don't need to store it, `web_contents()` is inherited from `WebContentsObserver`